server/src/server.ts

```diff
@@ -248,12 +249,24 @@ export class TcpClient implements ProtocolClient {
 		this.cryptoPromise = fetchHasJoined({
 			username: this.claimedMojangName,
 			shaHex,
-		}).then((mojangAuth) => {
+		}).then(async (mojangAuth) => {
 			if (!mojangAuth?.uuid) {
 				this.kick(`Mojang auth failed`)
 				throw new Error(`Mojang auth failed`)
 			}
-			this.log('Authenticated as', mojangAuth)
+
+			cache_uuid(this.claimedMojangName!, mojangAuth.uuid);
```

Comment:
claimedMojangName is client-provided and so may be mIsSpElLeD. we should use mojangAuth.name instead